spec/lib/workos/organizations_spec.rb

```diff
@@ -6,16 +6,80 @@
 
   describe '.create_organization' do
     context 'with valid payload' do
-      it 'creates an organization' do
-        VCR.use_cassette 'organization/create' do
-          organization = described_class.create_organization(
-            domains: ['example.io'],
-            name: 'Test Organization',
-          )
+      context 'with no idempotency key' do
+        it 'creates an organization' do
+          VCR.use_cassette 'organization/create' do
+            organization = described_class.create_organization(
+              domains: ['example.io'],
+              name: 'Test Organization',
+            )
 
-          expect(organization.id).to eq('org_01FCPEJXEZR4DSBA625YMGQT9N')
-          expect(organization.name).to eq('Test Organization')
-          expect(organization.domains.first[:domain]).to eq('example.io')
+            expect(organization.id).to eq('org_01FCPEJXEZR4DSBA625YMGQT9N')
+            expect(organization.name).to eq('Test Organization')
+            expect(organization.domains.first[:domain]).to eq('example.io')
+          end
+        end
+      end
+
+      context 'with idempotency key' do
+        context 'when idempotency key is used once' do
+          it 'creates an organization' do
+            VCR.use_cassette 'organization/create_with_idempotency_key' do
+              organization = described_class.create_organization(
+                domains: ['example.io'],
+                name: 'Test Organization',
+                idempotency_key: 'key',
+              )
+
+              expect(organization.name).to eq('Test Organization')
+              expect(organization.domains.first[:domain]).to eq('example.io')
+            end
+          end
+        end
+
+        context 'when idempotency key is used more than once' do
+          context 'with duplicate event payloads' do
+            it 'creates an organization' do
+              VCR.use_cassette 'organization/create_with_duplicate_idempotency_key_and_payload' do
+                organization1 = described_class.create_organization(
+                  domains: ['example.com'],
+                  name: 'Test Organization',
+                  idempotency_key: 'foo',
+                )
+
+                organization2 = described_class.create_organization(
```

Comment:
Can we also add a test here to make sure the count of `Organization` objects didn't go up? Maybe something like `assert_no_difference("Organization.count") {` around the second `create_organization` call?

Reply:
🤔, actually this one probably won't work, so never mind here - the Organization is only being created in our database, and the SDK is just consuming the API results, so it has no way of checking the count of organizations, so 👍 to just the `id` comparison you already had in there!

Reply:
Cool sounds good - I tried to find a way, but yeah couldn't do this one.

Reply:
Yeah, sorry if I'd sent you on a wild goose chase there :(

Reply:
No worries!